File: akuntansi_db/app/Models/PurchaseHeader.php
<?php

namespace App\Models;

use Carbon\Carbon;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class PurchaseHeader extends Model
{
    use HasFactory;

    protected $fillable = [
        'transaction_date',
        's_id',
        'gross_amount',
        'net_amount',
        'note',
    ];

    public function setTransactionDateAttribute($value)
    {
        $this->attributes['transaction_date'] = Carbon::parse($value);
    }
}

File: akuntansi_flut/lib/services/model/request/purchase_header.dart
class PurchaseHeaderRequest {
  String? transactionDate;
  String? sId;
  String? grossAmount;
  String? netAmount;
  String? note;

  PurchaseHeaderRequest({this.transactionDate, this.sId, this.grossAmount, this.netAmount, this.note});

  PurchaseHeaderRequest.fromJson(Map<String, dynamic> json) {
    transactionDate = json['transaction_date'];
    sId = json['s_id'];
    grossAmount = json['gross_amount'];
    netAmount = json['net_amount'];
    note = json['note'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['transaction_date'] = transactionDate;
    data['s_id'] = sId;
    data['gross_amount'] = grossAmount;
    data['net_amount'] = netAmount;
    data['note'] = note;
    return data;
  }
}
